Description: make prefix.freebsd* similar to prefix.linux
 The -msse4 flag is of particular interest, but also the -std=gnu++03
--- a/builds/posix/prefix.freebsd
+++ b/builds/posix/prefix.freebsd
@@ -18,7 +18,14 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major Cleanup
 
-EXE_LINK_OPTIONS+=-Wl,-rpath,../gen/firebird/lib
+COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DFREEBSD -pipe -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks
+CXXFLAGS+=-std=gnu++03
+OPTIMIZE_FLAGS=-O3 -march=i586 -mtune=i686 -fno-omit-frame-pointer
+WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-narrowing
 
-PROD_FLAGS=-O -fno-builtin -DFREEBSD -pipe -MMD -fPIC -std=gnu++03
-DEV_FLAGS=-ggdb -DFREEBSD -pipe -MMD -p -fPIC -Wall -Wno-non-virtual-dtor -std=gnu++03
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+#DEV_FLAGS=-DUSE_VALGRIND -p $(COMMON_FLAGS) $(WARN_FLAGS)
+DEV_FLAGS=-p $(COMMON_FLAGS) $(WARN_FLAGS)
+
+# This file must be compiled with SSE4.2 support
+%/CRC32C.o: COMMON_FLAGS += -msse4
--- a/builds/posix/prefix.freebsd_amd64
+++ b/builds/posix/prefix.freebsd_amd64
@@ -18,7 +18,14 @@
 #
 # 2 Oct 2002, Nickolay Samofatov - Major Cleanup
 
-EXE_LINK_OPTIONS+=-Wl,-rpath,../gen/firebird/lib
+COMMON_FLAGS=-ggdb -DFB_SEND_FLAGS=MSG_NOSIGNAL -DFREEBSD -DAMD64 -pipe -MMD -fPIC -fmessage-length=0 -fno-delete-null-pointer-checks
+CXXFLAGS+=-std=gnu++03
+OPTIMIZE_FLAGS=-O3 -fno-omit-frame-pointer
+WARN_FLAGS=-Wall -Wno-switch -Wno-parentheses -Wno-unknown-pragmas -Wno-unused-variable -Wno-narrowing -Wno-invalid-offsetof -Wno-unused-local-typedefs
 
-PROD_FLAGS=-O -fno-builtin -DFREEBSD -DAMD64 -pipe -MMD -fPIC -std=gnu++03
-DEV_FLAGS=-ggdb -DFREEBSD -DAMD64 -pipe -MMD -p -fPIC -Wall -Wno-non-virtual-dtor -std=gnu++03
+PROD_FLAGS=$(COMMON_FLAGS) $(OPTIMIZE_FLAGS)
+#DEV_FLAGS=-DUSE_VALGRIND -p $(COMMON_FLAGS) $(WARN_FLAGS)
+DEV_FLAGS=-p $(COMMON_FLAGS) $(WARN_FLAGS)
+
+# This file must be compiled with SSE4.2 support
+%/CRC32C.o: COMMON_FLAGS += -msse4
